src/main/java/glowredman/txloader/MinecraftClassTransformer.java

```diff
@@ -10,80 +8,52 @@
-
-        ClassNode classNode = new ClassNode();
-        ClassReader classReader = new ClassReader(basicClass);
-        classReader.accept(classNode, 0);
-
-        // find refreshResources() method
-        MethodNode targetMethod = null;
-        final String refreshResourcesName = devEnv ? "refreshResources" : "func_110436_a";
-        final String refreshResourcesDesc = "()V";
-
-        for (MethodNode method : classNode.methods) {
-            if (method.name.equals(refreshResourcesName) && method.desc.equals(refreshResourcesDesc)) {
-                targetMethod = method;
-                break;
-            }
-        }
-
-        if (targetMethod == null) throw new RuntimeException("Could not find method refreshResources()!");
-
-        // find first invocation of IReloadableResourceManager.reloadResources()
-        AbstractInsnNode targetInsn = null;
-        final String reloadResourcesName = devEnv ? "reloadResources" : "func_110541_a";
-        final String reloadResourcesDesc = "(Ljava/util/List;)V";
-
-        for (AbstractInsnNode ain : targetMethod.instructions.toArray()) {
-            if (ain instanceof MethodInsnNode) {
-                MethodInsnNode min = (MethodInsnNode) ain;
-                if (min.name.equals(reloadResourcesName) && min.desc.equals(reloadResourcesDesc)) {
-                    targetInsn = ain;
-                    break;
+        TXLoaderCore.LOGGER.debug("Transforming net.minecraft.client.Minecraft");
+        final boolean devEnv = (boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment");
+        final ClassNode classNode = new ClassNode();
+        final ClassReader classReader = new ClassReader(basicClass);
+        classReader.accept(classNode, ClassReader.SKIP_DEBUG);
+        final String targetMethodName = devEnv ? "refreshResources" : "func_110436_a";
+        final String targetMethodInsnName = devEnv ? "reloadResources" : "func_110541_a";
+        boolean success = false;
+        for (MethodNode mn : classNode.methods) {
+            if (mn.name.equals(targetMethodName) && mn.desc.equals("()V")) {
+                for (AbstractInsnNode node : mn.instructions.toArray()) {
+                    if (node instanceof MethodInsnNode && ((MethodInsnNode) node).name.equals(targetMethodInsnName)
+                            && ((MethodInsnNode) node).desc.equals("(Ljava/util/List;)V")) {
+                        mn.instructions.insertBefore(
+                                node,
+                                new MethodInsnNode(
+                                        Opcodes.INVOKESTATIC,
+                                        "glowredman/txloader/MinecraftHook",
+                                        "insertForcePack",
+                                        "(Ljava/util/List;)Ljava/util/List;",
+                                        false));
+                        success = true;
+                        break;
+                    }
```

Comment:
I feel like this whole section became less readable. No comments, no blank lines to separate code blocks, way more indentation. Also only one Exception at the end in case the transformer fails instead of multiple checks throughout, this makes it harder to find faults.

Reply:
now it's going straight to the point, before it would have too many lines to just find a method and a single instruction

Reply:
and there isn't much that can fail in this ASM, it can only fail of someone removes the targeted method call